// googleapis/build/src/apis/analyticsreporting/v4.js
"use strict";
/**
 * Copyright 2015 Google Inc. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
var apirequest_1 = require("../../lib/apirequest");
// TODO: We will eventually get the `any` in here cleared out, but in the
// interim we want to turn on no-implicit-any.
// tslint:disable: no-any
// tslint:disable: class-name
// tslint:disable: variable-name
// tslint:disable: jsdoc-format
// tslint:disable: no-namespace
var analyticsreporting_v4;
(function (analyticsreporting_v4) {
    /**
     * Google Analytics Reporting API
     *
     * Accesses Analytics report data.
     *
     * @example
     * const google = require('googleapis');
     * const analyticsreporting = google.analyticsreporting('v4');
     *
     * @namespace analyticsreporting
     * @type {Function}
     * @version v4
     * @variation v4
     * @param {object=} options Options for Analyticsreporting
     */
    var Analyticsreporting = /** @class */ (function () {
        function Analyticsreporting(options, google) {
            this.root = this;
            this._options = options || {};
            this.google = google;
            this.getRoot.bind(this);
            this.reports = new Resource$Reports(this);
        }
        Analyticsreporting.prototype.getRoot = function () {
            return this.root;
        };
        return Analyticsreporting;
    }());
    analyticsreporting_v4.Analyticsreporting = Analyticsreporting;
    var Resource$Reports = /** @class */ (function () {
        function Resource$Reports(root) {
            this.root = root;
            this.getRoot.bind(this);
        }
        Resource$Reports.prototype.getRoot = function () {
            return this.root;
        };
        Resource$Reports.prototype.batchGet = function (paramsOrCallback, optionsOrCallback, callback) {
            var params = (paramsOrCallback || {});
            var options = (optionsOrCallback || {});
            if (typeof paramsOrCallback === 'function') {
                callback = paramsOrCallback;
                params = {};
                options = {};
            }
            if (typeof optionsOrCallback === 'function') {
                callback = optionsOrCallback;
                options = {};
            }
            var rootUrl = options.rootUrl || 'https://analyticsreporting.googleapis.com/';
            var parameters = {
                options: Object.assign({
                    url: (rootUrl + '/v4/reports:batchGet')
                        .replace(/([^:]\/)\/+/g, '$1'),
                    method: 'POST'
                }, options),
                params: params,
                requiredParams: [],
                pathParams: [],
                context: this.getRoot()
            };
            if (callback) {
                apirequest_1.createAPIRequest(parameters, callback);
            }
            else {
                return apirequest_1.createAPIRequest(parameters);
            }
        };
        return Resource$Reports;
    }());
    analyticsreporting_v4.Resource$Reports = Resource$Reports;
})(analyticsreporting_v4 = exports.analyticsreporting_v4 || (exports.analyticsreporting_v4 = {}));
