// googleapis/build/src/apis/servicebroker/v1alpha1.js
"use strict";
/**
 * Copyright 2015 Google Inc. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
var apirequest_1 = require("../../lib/apirequest");
// TODO: We will eventually get the `any` in here cleared out, but in the
// interim we want to turn on no-implicit-any.
// tslint:disable: no-any
// tslint:disable: class-name
// tslint:disable: variable-name
// tslint:disable: jsdoc-format
// tslint:disable: no-namespace
var servicebroker_v1alpha1;
(function (servicebroker_v1alpha1) {
    /**
     * Service Broker API
     *
     * The Google Cloud Platform Service Broker API provides Google hosted
     * implementation of the Open Service Broker API
     * (https://www.openservicebrokerapi.org/).
     *
     * @example
     * const google = require('googleapis');
     * const servicebroker = google.servicebroker('v1alpha1');
     *
     * @namespace servicebroker
     * @type {Function}
     * @version v1alpha1
     * @variation v1alpha1
     * @param {object=} options Options for Servicebroker
     */
    var Servicebroker = /** @class */ (function () {
        function Servicebroker(options, google) {
            this.root = this;
            this._options = options || {};
            this.google = google;
            this.getRoot.bind(this);
            this.projects = new Resource$Projects(this);
            this.v1alpha1 = new Resource$V1alpha1(this);
        }
        Servicebroker.prototype.getRoot = function () {
            return this.root;
        };
        return Servicebroker;
    }());
    servicebroker_v1alpha1.Servicebroker = Servicebroker;
    var Resource$Projects = /** @class */ (function () {
        function Resource$Projects(root) {
            this.root = root;
            this.getRoot.bind(this);
            this.brokers = new Resource$Projects$Brokers(root);
        }
        Resource$Projects.prototype.getRoot = function () {
            return this.root;
        };
        return Resource$Projects;
    }());
    servicebroker_v1alpha1.Resource$Projects = Resource$Projects;
    var Resource$Projects$Brokers = /** @class */ (function () {
        function Resource$Projects$Brokers(root) {
            this.root = root;
            this.getRoot.bind(this);
            this.instances = new Resource$Projects$Brokers$Instances(root);
            this.service_instances =
                new Resource$Projects$Brokers$Service_instances(root);
            this.v2 = new Resource$Projects$Brokers$V2(root);
        }
        Resource$Projects$Brokers.prototype.getRoot = function () {
            return this.root;
        };
        return Resource$Projects$Brokers;
    }());
    servicebroker_v1alpha1.Resource$Projects$Brokers = Resource$Projects$Brokers;
    var Resource$Projects$Brokers$Instances = /** @class */ (function () {
        function Resource$Projects$Brokers$Instances(root) {
            this.root = root;
            this.getRoot.bind(this);
            this.service_bindings =
                new Resource$Projects$Brokers$Instances$Service_bindings(root);
        }
        Resource$Projects$Brokers$Instances.prototype.getRoot = function () {
            return this.root;
        };
        Resource$Projects$Brokers$Instances.prototype.get = function (paramsOrCallback, optionsOrCallback, callback) {
            var params = (paramsOrCallback || {});
            var options = (optionsOrCallback || {});
            if (typeof paramsOrCallback === 'function') {
                callback = paramsOrCallback;
                params = {};
                options = {};
            }
            if (typeof optionsOrCallback === 'function') {
                callback = optionsOrCallback;
                options = {};
            }
            var rootUrl = options.rootUrl || 'https://servicebroker.googleapis.com/';
            var parameters = {
                options: Object.assign({
                    url: (rootUrl + '/v1alpha1/{+name}').replace(/([^:]\/)\/+/g, '$1'),
                    method: 'GET'
                }, options),
                params: params,
                requiredParams: ['name'],
                pathParams: ['name'],
                context: this.getRoot()
            };
            if (callback) {
                apirequest_1.createAPIRequest(parameters, callback);
            }
            else {
                return apirequest_1.createAPIRequest(parameters);
            }
        };
        return Resource$Projects$Brokers$Instances;
    }());
    servicebroker_v1alpha1.Resource$Projects$Brokers$Instances = Resource$Projects$Brokers$Instances;
    var Resource$Projects$Brokers$Instances$Service_bindings = /** @class */ (function () {
        function Resource$Projects$Brokers$Instances$Service_bindings(root) {
            this.root = root;
            this.getRoot.bind(this);
        }
        Resource$Projects$Brokers$Instances$Service_bindings.prototype.getRoot = function () {
            return this.root;
        };
        Resource$Projects$Brokers$Instances$Service_bindings.prototype.list = function (paramsOrCallback, optionsOrCallback, callback) {
            var params = (paramsOrCallback || {});
            var options = (optionsOrCallback || {});
            if (typeof paramsOrCallback === 'function') {
                callback = paramsOrCallback;
                params = {};
                options = {};
            }
            if (typeof optionsOrCallback === 'function') {
                callback = optionsOrCallback;
                options = {};
            }
            var rootUrl = options.rootUrl || 'https://servicebroker.googleapis.com/';
            var parameters = {
                options: Object.assign({
                    url: (rootUrl + '/v1alpha1/{+parent}/service_bindings')
                        .replace(/([^:]\/)\/+/g, '$1'),
                    method: 'GET'
                }, options),
                params: params,
                requiredParams: ['parent'],
                pathParams: ['parent'],
                context: this.getRoot()
            };
            if (callback) {
                apirequest_1.createAPIRequest(parameters, callback);
            }
            else {
                return apirequest_1.createAPIRequest(parameters);
            }
        };
        return Resource$Projects$Brokers$Instances$Service_bindings;
    }());
    servicebroker_v1alpha1.Resource$Projects$Brokers$Instances$Service_bindings = Resource$Projects$Brokers$Instances$Service_bindings;
    var Resource$Projects$Brokers$Service_instances = /** @class */ (function () {
        function Resource$Projects$Brokers$Service_instances(root) {
            this.root = root;
            this.getRoot.bind(this);
        }
        Resource$Projects$Brokers$Service_instances.prototype.getRoot = function () {
            return this.root;
        };
        Resource$Projects$Brokers$Service_instances.prototype.list = function (paramsOrCallback, optionsOrCallback, callback) {
            var params = (paramsOrCallback || {});
            var options = (optionsOrCallback || {});
            if (typeof paramsOrCallback === 'function') {
                callback = paramsOrCallback;
                params = {};
                options = {};
            }
            if (typeof optionsOrCallback === 'function') {
                callback = optionsOrCallback;
                options = {};
            }
            var rootUrl = options.rootUrl || 'https://servicebroker.googleapis.com/';
            var parameters = {
                options: Object.assign({
                    url: (rootUrl + '/v1alpha1/{+parent}/service_instances')
                        .replace(/([^:]\/)\/+/g, '$1'),
                    method: 'GET'
                }, options),
                params: params,
                requiredParams: ['parent'],
                pathParams: ['parent'],
                context: this.getRoot()
            };
            if (callback) {
                apirequest_1.createAPIRequest(parameters, callback);
            }
            else {
                return apirequest_1.createAPIRequest(parameters);
            }
        };
        return Resource$Projects$Brokers$Service_instances;
    }());
    servicebroker_v1alpha1.Resource$Projects$Brokers$Service_instances = Resource$Projects$Brokers$Service_instances;
    var Resource$Projects$Brokers$V2 = /** @class */ (function () {
        function Resource$Projects$Brokers$V2(root) {
            this.root = root;
            this.getRoot.bind(this);
            this.catalog = new Resource$Projects$Brokers$V2$Catalog(root);
            this.service_instances =
                new Resource$Projects$Brokers$V2$Service_instances(root);
        }
        Resource$Projects$Brokers$V2.prototype.getRoot = function () {
            return this.root;
        };
        return Resource$Projects$Brokers$V2;
    }());
    servicebroker_v1alpha1.Resource$Projects$Brokers$V2 = Resource$Projects$Brokers$V2;
    var Resource$Projects$Brokers$V2$Catalog = /** @class */ (function () {
        function Resource$Projects$Brokers$V2$Catalog(root) {
            this.root = root;
            this.getRoot.bind(this);
        }
        Resource$Projects$Brokers$V2$Catalog.prototype.getRoot = function () {
            return this.root;
        };
        Resource$Projects$Brokers$V2$Catalog.prototype.list = function (paramsOrCallback, optionsOrCallback, callback) {
            var params = (paramsOrCallback || {});
            var options = (optionsOrCallback || {});
            if (typeof paramsOrCallback === 'function') {
                callback = paramsOrCallback;
                params = {};
                options = {};
            }
            if (typeof optionsOrCallback === 'function') {
                callback = optionsOrCallback;
                options = {};
            }
            var rootUrl = options.rootUrl || 'https://servicebroker.googleapis.com/';
            var parameters = {
                options: Object.assign({
                    url: (rootUrl + '/v1alpha1/{+parent}/v2/catalog')
                        .replace(/([^:]\/)\/+/g, '$1'),
                    method: 'GET'
                }, options),
                params: params,
                requiredParams: ['parent'],
                pathParams: ['parent'],
                context: this.getRoot()
            };
            if (callback) {
                apirequest_1.createAPIRequest(parameters, callback);
            }
            else {
                return apirequest_1.createAPIRequest(parameters);
            }
        };
        return Resource$Projects$Brokers$V2$Catalog;
    }());
    servicebroker_v1alpha1.Resource$Projects$Brokers$V2$Catalog = Resource$Projects$Brokers$V2$Catalog;
    var Resource$Projects$Brokers$V2$Service_instances = /** @class */ (function () {
        function Resource$Projects$Brokers$V2$Service_instances(root) {
            this.root = root;
            this.getRoot.bind(this);
            this.service_bindings =
                new Resource$Projects$Brokers$V2$Service_instances$Service_bindings(root);
        }
        Resource$Projects$Brokers$V2$Service_instances.prototype.getRoot = function () {
            return this.root;
        };
        Resource$Projects$Brokers$V2$Service_instances.prototype.create = function (paramsOrCallback, optionsOrCallback, callback) {
            var params = (paramsOrCallback || {});
            var options = (optionsOrCallback || {});
            if (typeof paramsOrCallback === 'function') {
                callback = paramsOrCallback;
                params = {};
                options = {};
            }
            if (typeof optionsOrCallback === 'function') {
                callback = optionsOrCallback;
                options = {};
            }
            var rootUrl = options.rootUrl || 'https://servicebroker.googleapis.com/';
            var parameters = {
                options: Object.assign({
                    url: (rootUrl +
                        '/v1alpha1/{+parent}/v2/service_instances/{+instance_id}')
                        .replace(/([^:]\/)\/+/g, '$1'),
                    method: 'PUT'
                }, options),
                params: params,
                requiredParams: ['parent', 'instance_id'],
                pathParams: ['instance_id', 'parent'],
                context: this.getRoot()
            };
            if (callback) {
                apirequest_1.createAPIRequest(parameters, callback);
            }
            else {
                return apirequest_1.createAPIRequest(parameters);
            }
        };
        Resource$Projects$Brokers$V2$Service_instances.prototype.delete = function (paramsOrCallback, optionsOrCallback, callback) {
            var params = (paramsOrCallback || {});
            var options = (optionsOrCallback || {});
            if (typeof paramsOrCallback === 'function') {
                callback = paramsOrCallback;
                params = {};
                options = {};
            }
            if (typeof optionsOrCallback === 'function') {
                callback = optionsOrCallback;
                options = {};
            }
            var rootUrl = options.rootUrl || 'https://servicebroker.googleapis.com/';
            var parameters = {
                options: Object.assign({
                    url: (rootUrl +
                        '/v1alpha1/{+parent}/v2/service_instances/{+instanceId}')
                        .replace(/([^:]\/)\/+/g, '$1'),
                    method: 'DELETE'
                }, options),
                params: params,
                requiredParams: ['parent', 'instanceId'],
                pathParams: ['instanceId', 'parent'],
                context: this.getRoot()
            };
            if (callback) {
                apirequest_1.createAPIRequest(parameters, callback);
            }
            else {
                return apirequest_1.createAPIRequest(parameters);
            }
        };
        Resource$Projects$Brokers$V2$Service_instances.prototype.get = function (paramsOrCallback, optionsOrCallback, callback) {
            var params = (paramsOrCallback || {});
            var options = (optionsOrCallback || {});
            if (typeof paramsOrCallback === 'function') {
                callback = paramsOrCallback;
                params = {};
                options = {};
            }
            if (typeof optionsOrCallback === 'function') {
                callback = optionsOrCallback;
                options = {};
            }
            var rootUrl = options.rootUrl || 'https://servicebroker.googleapis.com/';
            var parameters = {
                options: Object.assign({
                    url: (rootUrl + '/v1alpha1/{+name}').replace(/([^:]\/)\/+/g, '$1'),
                    method: 'GET'
                }, options),
                params: params,
                requiredParams: ['name'],
                pathParams: ['name'],
                context: this.getRoot()
            };
            if (callback) {
                apirequest_1.createAPIRequest(parameters, callback);
            }
            else {
                return apirequest_1.createAPIRequest(parameters);
            }
        };
        Resource$Projects$Brokers$V2$Service_instances.prototype.getLast_operation = function (paramsOrCallback, optionsOrCallback, callback) {
            var params = (paramsOrCallback || {});
            var options = (optionsOrCallback || {});
            if (typeof paramsOrCallback === 'function') {
                callback = paramsOrCallback;
                params = {};
                options = {};
            }
            if (typeof optionsOrCallback === 'function') {
                callback = optionsOrCallback;
                options = {};
            }
            var rootUrl = options.rootUrl || 'https://servicebroker.googleapis.com/';
            var parameters = {
                options: Object.assign({
                    url: (rootUrl +
                        '/v1alpha1/{+parent}/v2/service_instances/{+instanceId}/last_operation')
                        .replace(/([^:]\/)\/+/g, '$1'),
                    method: 'GET'
                }, options),
                params: params,
                requiredParams: ['parent', 'instanceId'],
                pathParams: ['instanceId', 'parent'],
                context: this.getRoot()
            };
            if (callback) {
                apirequest_1.createAPIRequest(parameters, callback);
            }
            else {
                return apirequest_1.createAPIRequest(parameters);
            }
        };
        Resource$Projects$Brokers$V2$Service_instances.prototype.patch = function (paramsOrCallback, optionsOrCallback, callback) {
            var params = (paramsOrCallback || {});
            var options = (optionsOrCallback || {});
            if (typeof paramsOrCallback === 'function') {
                callback = paramsOrCallback;
                params = {};
                options = {};
            }
            if (typeof optionsOrCallback === 'function') {
                callback = optionsOrCallback;
                options = {};
            }
            var rootUrl = options.rootUrl || 'https://servicebroker.googleapis.com/';
            var parameters = {
                options: Object.assign({
                    url: (rootUrl +
                        '/v1alpha1/{+parent}/v2/service_instances/{+instance_id}')
                        .replace(/([^:]\/)\/+/g, '$1'),
                    method: 'PATCH'
                }, options),
                params: params,
                requiredParams: ['parent', 'instance_id'],
                pathParams: ['instance_id', 'parent'],
                context: this.getRoot()
            };
            if (callback) {
                apirequest_1.createAPIRequest(parameters, callback);
            }
            else {
                return apirequest_1.createAPIRequest(parameters);
            }
        };
        return Resource$Projects$Brokers$V2$Service_instances;
    }());
    servicebroker_v1alpha1.Resource$Projects$Brokers$V2$Service_instances = Resource$Projects$Brokers$V2$Service_instances;
    var Resource$Projects$Brokers$V2$Service_instances$Service_bindings = /** @class */ (function () {
        function Resource$Projects$Brokers$V2$Service_instances$Service_bindings(root) {
            this.root = root;
            this.getRoot.bind(this);
        }
        Resource$Projects$Brokers$V2$Service_instances$Service_bindings.prototype.getRoot = function () {
            return this.root;
        };
        Resource$Projects$Brokers$V2$Service_instances$Service_bindings.prototype.create = function (paramsOrCallback, optionsOrCallback, callback) {
            var params = (paramsOrCallback || {});
            var options = (optionsOrCallback || {});
            if (typeof paramsOrCallback === 'function') {
                callback = paramsOrCallback;
                params = {};
                options = {};
            }
            if (typeof optionsOrCallback === 'function') {
                callback = optionsOrCallback;
                options = {};
            }
            var rootUrl = options.rootUrl || 'https://servicebroker.googleapis.com/';
            var parameters = {
                options: Object.assign({
                    url: (rootUrl +
                        '/v1alpha1/{+parent}/v2/service_instances/{+instanceId}/service_bindings/{+binding_id}')
                        .replace(/([^:]\/)\/+/g, '$1'),
                    method: 'PUT'
                }, options),
                params: params,
                requiredParams: ['parent', 'instanceId', 'binding_id'],
                pathParams: ['binding_id', 'instanceId', 'parent'],
                context: this.getRoot()
            };
            if (callback) {
                apirequest_1.createAPIRequest(parameters, callback);
            }
            else {
                return apirequest_1.createAPIRequest(parameters);
            }
        };
        Resource$Projects$Brokers$V2$Service_instances$Service_bindings.prototype.delete = function (paramsOrCallback, optionsOrCallback, callback) {
            var params = (paramsOrCallback || {});
            var options = (optionsOrCallback || {});
            if (typeof paramsOrCallback === 'function') {
                callback = paramsOrCallback;
                params = {};
                options = {};
            }
            if (typeof optionsOrCallback === 'function') {
                callback = optionsOrCallback;
                options = {};
            }
            var rootUrl = options.rootUrl || 'https://servicebroker.googleapis.com/';
            var parameters = {
                options: Object.assign({
                    url: (rootUrl +
                        '/v1alpha1/{+parent}/v2/service_instances/{instanceId}/service_bindings/{bindingId}')
                        .replace(/([^:]\/)\/+/g, '$1'),
                    method: 'DELETE'
                }, options),
                params: params,
                requiredParams: ['parent', 'instanceId', 'bindingId'],
                pathParams: ['bindingId', 'instanceId', 'parent'],
                context: this.getRoot()
            };
            if (callback) {
                apirequest_1.createAPIRequest(parameters, callback);
            }
            else {
                return apirequest_1.createAPIRequest(parameters);
            }
        };
        Resource$Projects$Brokers$V2$Service_instances$Service_bindings.prototype.get = function (paramsOrCallback, optionsOrCallback, callback) {
            var params = (paramsOrCallback || {});
            var options = (optionsOrCallback || {});
            if (typeof paramsOrCallback === 'function') {
                callback = paramsOrCallback;
                params = {};
                options = {};
            }
            if (typeof optionsOrCallback === 'function') {
                callback = optionsOrCallback;
                options = {};
            }
            var rootUrl = options.rootUrl || 'https://servicebroker.googleapis.com/';
            var parameters = {
                options: Object.assign({
                    url: (rootUrl +
                        '/v1alpha1/{+parent}/v2/service_instances/{+instanceId}/service_bindings/{+bindingId}')
                        .replace(/([^:]\/)\/+/g, '$1'),
                    method: 'GET'
                }, options),
                params: params,
                requiredParams: ['parent', 'instanceId', 'bindingId'],
                pathParams: ['bindingId', 'instanceId', 'parent'],
                context: this.getRoot()
            };
            if (callback) {
                apirequest_1.createAPIRequest(parameters, callback);
            }
            else {
                return apirequest_1.createAPIRequest(parameters);
            }
        };
        Resource$Projects$Brokers$V2$Service_instances$Service_bindings.prototype.getLast_operation = function (paramsOrCallback, optionsOrCallback, callback) {
            var params = (paramsOrCallback || {});
            var options = (optionsOrCallback || {});
            if (typeof paramsOrCallback === 'function') {
                callback = paramsOrCallback;
                params = {};
                options = {};
            }
            if (typeof optionsOrCallback === 'function') {
                callback = optionsOrCallback;
                options = {};
            }
            var rootUrl = options.rootUrl || 'https://servicebroker.googleapis.com/';
            var parameters = {
                options: Object.assign({
                    url: (rootUrl +
                        '/v1alpha1/{+parent}/v2/service_instances/{+instanceId}/service_bindings/{+bindingId}/last_operation')
                        .replace(/([^:]\/)\/+/g, '$1'),
                    method: 'GET'
                }, options),
                params: params,
                requiredParams: ['parent', 'instanceId', 'bindingId'],
                pathParams: ['bindingId', 'instanceId', 'parent'],
                context: this.getRoot()
            };
            if (callback) {
                apirequest_1.createAPIRequest(parameters, callback);
            }
            else {
                return apirequest_1.createAPIRequest(parameters);
            }
        };
        return Resource$Projects$Brokers$V2$Service_instances$Service_bindings;
    }());
    servicebroker_v1alpha1.Resource$Projects$Brokers$V2$Service_instances$Service_bindings = Resource$Projects$Brokers$V2$Service_instances$Service_bindings;
    var Resource$V1alpha1 = /** @class */ (function () {
        function Resource$V1alpha1(root) {
            this.root = root;
            this.getRoot.bind(this);
        }
        Resource$V1alpha1.prototype.getRoot = function () {
            return this.root;
        };
        Resource$V1alpha1.prototype.getIamPolicy = function (paramsOrCallback, optionsOrCallback, callback) {
            var params = (paramsOrCallback || {});
            var options = (optionsOrCallback || {});
            if (typeof paramsOrCallback === 'function') {
                callback = paramsOrCallback;
                params = {};
                options = {};
            }
            if (typeof optionsOrCallback === 'function') {
                callback = optionsOrCallback;
                options = {};
            }
            var rootUrl = options.rootUrl || 'https://servicebroker.googleapis.com/';
            var parameters = {
                options: Object.assign({
                    url: (rootUrl + '/v1alpha1/{+resource}:getIamPolicy')
                        .replace(/([^:]\/)\/+/g, '$1'),
                    method: 'GET'
                }, options),
                params: params,
                requiredParams: ['resource'],
                pathParams: ['resource'],
                context: this.getRoot()
            };
            if (callback) {
                apirequest_1.createAPIRequest(parameters, callback);
            }
            else {
                return apirequest_1.createAPIRequest(parameters);
            }
        };
        Resource$V1alpha1.prototype.setIamPolicy = function (paramsOrCallback, optionsOrCallback, callback) {
            var params = (paramsOrCallback || {});
            var options = (optionsOrCallback || {});
            if (typeof paramsOrCallback === 'function') {
                callback = paramsOrCallback;
                params = {};
                options = {};
            }
            if (typeof optionsOrCallback === 'function') {
                callback = optionsOrCallback;
                options = {};
            }
            var rootUrl = options.rootUrl || 'https://servicebroker.googleapis.com/';
            var parameters = {
                options: Object.assign({
                    url: (rootUrl + '/v1alpha1/{+resource}:setIamPolicy')
                        .replace(/([^:]\/)\/+/g, '$1'),
                    method: 'POST'
                }, options),
                params: params,
                requiredParams: ['resource'],
                pathParams: ['resource'],
                context: this.getRoot()
            };
            if (callback) {
                apirequest_1.createAPIRequest(parameters, callback);
            }
            else {
                return apirequest_1.createAPIRequest(parameters);
            }
        };
        Resource$V1alpha1.prototype.testIamPermissions = function (paramsOrCallback, optionsOrCallback, callback) {
            var params = (paramsOrCallback || {});
            var options = (optionsOrCallback || {});
            if (typeof paramsOrCallback === 'function') {
                callback = paramsOrCallback;
                params = {};
                options = {};
            }
            if (typeof optionsOrCallback === 'function') {
                callback = optionsOrCallback;
                options = {};
            }
            var rootUrl = options.rootUrl || 'https://servicebroker.googleapis.com/';
            var parameters = {
                options: Object.assign({
                    url: (rootUrl + '/v1alpha1/{+resource}:testIamPermissions')
                        .replace(/([^:]\/)\/+/g, '$1'),
                    method: 'POST'
                }, options),
                params: params,
                requiredParams: ['resource'],
                pathParams: ['resource'],
                context: this.getRoot()
            };
            if (callback) {
                apirequest_1.createAPIRequest(parameters, callback);
            }
            else {
                return apirequest_1.createAPIRequest(parameters);
            }
        };
        return Resource$V1alpha1;
    }());
    servicebroker_v1alpha1.Resource$V1alpha1 = Resource$V1alpha1;
})(servicebroker_v1alpha1 = exports.servicebroker_v1alpha1 || (exports.servicebroker_v1alpha1 = {}));
