// googleapis/build/src/apis/firebasedynamiclinks/v1.js
"use strict";
/**
 * Copyright 2015 Google Inc. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
var apirequest_1 = require("../../lib/apirequest");
// TODO: We will eventually get the `any` in here cleared out, but in the
// interim we want to turn on no-implicit-any.
// tslint:disable: no-any
// tslint:disable: class-name
// tslint:disable: variable-name
// tslint:disable: jsdoc-format
// tslint:disable: no-namespace
var firebasedynamiclinks_v1;
(function (firebasedynamiclinks_v1) {
    /**
     * Firebase Dynamic Links API
     *
     * Programmatically creates and manages Firebase Dynamic Links.
     *
     * @example
     * const google = require('googleapis');
     * const firebasedynamiclinks = google.firebasedynamiclinks('v1');
     *
     * @namespace firebasedynamiclinks
     * @type {Function}
     * @version v1
     * @variation v1
     * @param {object=} options Options for Firebasedynamiclinks
     */
    var Firebasedynamiclinks = /** @class */ (function () {
        function Firebasedynamiclinks(options, google) {
            this.root = this;
            this._options = options || {};
            this.google = google;
            this.getRoot.bind(this);
            this.managedShortLinks = new Resource$Managedshortlinks(this);
            this.shortLinks = new Resource$Shortlinks(this);
            this.v1 = new Resource$V1(this);
        }
        Firebasedynamiclinks.prototype.getRoot = function () {
            return this.root;
        };
        return Firebasedynamiclinks;
    }());
    firebasedynamiclinks_v1.Firebasedynamiclinks = Firebasedynamiclinks;
    var Resource$Managedshortlinks = /** @class */ (function () {
        function Resource$Managedshortlinks(root) {
            this.root = root;
            this.getRoot.bind(this);
        }
        Resource$Managedshortlinks.prototype.getRoot = function () {
            return this.root;
        };
        Resource$Managedshortlinks.prototype.create = function (paramsOrCallback, optionsOrCallback, callback) {
            var params = (paramsOrCallback || {});
            var options = (optionsOrCallback || {});
            if (typeof paramsOrCallback === 'function') {
                callback = paramsOrCallback;
                params = {};
                options = {};
            }
            if (typeof optionsOrCallback === 'function') {
                callback = optionsOrCallback;
                options = {};
            }
            var rootUrl = options.rootUrl || 'https://firebasedynamiclinks.googleapis.com/';
            var parameters = {
                options: Object.assign({
                    url: (rootUrl + '/v1/managedShortLinks:create')
                        .replace(/([^:]\/)\/+/g, '$1'),
                    method: 'POST'
                }, options),
                params: params,
                requiredParams: [],
                pathParams: [],
                context: this.getRoot()
            };
            if (callback) {
                apirequest_1.createAPIRequest(parameters, callback);
            }
            else {
                return apirequest_1.createAPIRequest(parameters);
            }
        };
        return Resource$Managedshortlinks;
    }());
    firebasedynamiclinks_v1.Resource$Managedshortlinks = Resource$Managedshortlinks;
    var Resource$Shortlinks = /** @class */ (function () {
        function Resource$Shortlinks(root) {
            this.root = root;
            this.getRoot.bind(this);
        }
        Resource$Shortlinks.prototype.getRoot = function () {
            return this.root;
        };
        Resource$Shortlinks.prototype.create = function (paramsOrCallback, optionsOrCallback, callback) {
            var params = (paramsOrCallback || {});
            var options = (optionsOrCallback || {});
            if (typeof paramsOrCallback === 'function') {
                callback = paramsOrCallback;
                params = {};
                options = {};
            }
            if (typeof optionsOrCallback === 'function') {
                callback = optionsOrCallback;
                options = {};
            }
            var rootUrl = options.rootUrl || 'https://firebasedynamiclinks.googleapis.com/';
            var parameters = {
                options: Object.assign({
                    url: (rootUrl + '/v1/shortLinks').replace(/([^:]\/)\/+/g, '$1'),
                    method: 'POST'
                }, options),
                params: params,
                requiredParams: [],
                pathParams: [],
                context: this.getRoot()
            };
            if (callback) {
                apirequest_1.createAPIRequest(parameters, callback);
            }
            else {
                return apirequest_1.createAPIRequest(parameters);
            }
        };
        return Resource$Shortlinks;
    }());
    firebasedynamiclinks_v1.Resource$Shortlinks = Resource$Shortlinks;
    var Resource$V1 = /** @class */ (function () {
        function Resource$V1(root) {
            this.root = root;
            this.getRoot.bind(this);
        }
        Resource$V1.prototype.getRoot = function () {
            return this.root;
        };
        Resource$V1.prototype.getLinkStats = function (paramsOrCallback, optionsOrCallback, callback) {
            var params = (paramsOrCallback || {});
            var options = (optionsOrCallback || {});
            if (typeof paramsOrCallback === 'function') {
                callback = paramsOrCallback;
                params = {};
                options = {};
            }
            if (typeof optionsOrCallback === 'function') {
                callback = optionsOrCallback;
                options = {};
            }
            var rootUrl = options.rootUrl || 'https://firebasedynamiclinks.googleapis.com/';
            var parameters = {
                options: Object.assign({
                    url: (rootUrl + '/v1/{dynamicLink}/linkStats')
                        .replace(/([^:]\/)\/+/g, '$1'),
                    method: 'GET'
                }, options),
                params: params,
                requiredParams: ['dynamicLink'],
                pathParams: ['dynamicLink'],
                context: this.getRoot()
            };
            if (callback) {
                apirequest_1.createAPIRequest(parameters, callback);
            }
            else {
                return apirequest_1.createAPIRequest(parameters);
            }
        };
        Resource$V1.prototype.installAttribution = function (paramsOrCallback, optionsOrCallback, callback) {
            var params = (paramsOrCallback || {});
            var options = (optionsOrCallback || {});
            if (typeof paramsOrCallback === 'function') {
                callback = paramsOrCallback;
                params = {};
                options = {};
            }
            if (typeof optionsOrCallback === 'function') {
                callback = optionsOrCallback;
                options = {};
            }
            var rootUrl = options.rootUrl || 'https://firebasedynamiclinks.googleapis.com/';
            var parameters = {
                options: Object.assign({
                    url: (rootUrl + '/v1/installAttribution')
                        .replace(/([^:]\/)\/+/g, '$1'),
                    method: 'POST'
                }, options),
                params: params,
                requiredParams: [],
                pathParams: [],
                context: this.getRoot()
            };
            if (callback) {
                apirequest_1.createAPIRequest(parameters, callback);
            }
            else {
                return apirequest_1.createAPIRequest(parameters);
            }
        };
        return Resource$V1;
    }());
    firebasedynamiclinks_v1.Resource$V1 = Resource$V1;
})(firebasedynamiclinks_v1 = exports.firebasedynamiclinks_v1 || (exports.firebasedynamiclinks_v1 = {}));
